components/io_expander/esp_io_expander/esp_io_expander.c

```diff
@@ -14,9 +14,10 @@
 #include "esp_io_expander.h"
 
 /* Pin num must be smaller than 32 (bit width of uint32_t) */
-#define PIN_NUM_MAX(handle)                         (handle->config.io_amount > 32 ? 32 : handle->config.io_amount)
+#define PIN_NUM_MAX(handle)                 (((handle)->config.io_amount < IO_EXPANDER_PIN_NUM_MAX) ?  \
+                                             (handle)->config.io_amount : (IO_EXPANDER_PIN_NUM_MAX - 1))
```

Comment:
Is the calculation correct here? Why can't I set io_amount to 32?

Reply:
It's wrong. Fixed.

Reply:
@tore-espressif PTAL the new commit. Thank you very much.